pkg/controllers/workloadrebalancer/workloadrebalancer_controller.go

```diff
@@ -67,24 +67,25 @@ func (c *RebalancerController) Reconcile(ctx context.Context, req controllerrunt
 	klog.V(4).Infof("Reconciling for WorkloadRebalancer %s", req.Name)
 
 	// 1. get latest WorkloadRebalancer
-	rebalancer := &appsv1alpha1.WorkloadRebalancer{}
-	if err := c.Client.Get(ctx, req.NamespacedName, rebalancer); err != nil {
+	oldRebalancer := &appsv1alpha1.WorkloadRebalancer{}
+	if err := c.Client.Get(ctx, req.NamespacedName, oldRebalancer); err != nil {
 		if apierrors.IsNotFound(err) {
 			klog.Infof("no need to reconcile WorkloadRebalancer for it not found")
 			return controllerruntime.Result{}, nil
 		}
 		return controllerruntime.Result{}, err
 	}
+	rebalancer := oldRebalancer.DeepCopy()
 
 	// 2. get and update referenced binding to trigger a rescheduling
-	newStatus, successNum, retryNum := c.doWorkloadRebalance(ctx, rebalancer)
+	successNum, retryNum := c.doWorkloadRebalance(ctx, rebalancer)
```

Comment:
Not a good pattern to mutate the input parameters, as the method behavior is hard to predict.